### ISSUE TITLE
[2.0.1] Tag Select does not work with using and custom query on form

### ISSUE THREAD

**Issue by vikilaboy** [phalcon/cphalcon#10284]
Hey guys,

i have the code for selecting data:
```php
    public static function getForDropdown($status = self::STATUS_ACTIVE)
    {
        $builder = parent::getBuilder();

        return
            $builder->from(['a' => 'Rinf\Models\Employee'])
            ->columns(['a.id', 'CONCAT_WS(\' \', a.firstname, a.lastname) AS name'])
            ->where('a.status = :status:', ['status' => (int)$status])->orderBy('name ASC')
            ->getQuery()
            ->execute();
    }

```
and I call this in the form 
```php
        $employee = new Select('idEmployee', Employee::getForDropdown(),
            [
                'using' => ['id', 'name'],
                'useEmpty' => true,
                'emptyText' => t('Choose employee...')
            ]);
        $employee->setLabel(t('Employee'));
        $this->add($employee);
```
It renders the select tag with the number of rows but empty values and text in option for select tag.
This worked fine on 1.3.5

**Comment by GBraL** [comment]
Hello, I wore this way too, but after I updated to version 2.0.0 I had to switch and use a simple query "SELECT name FROM table ...".

The following example:

```php
		$phql = "
			SELECT
				permissao.id_permissao,
				CONCAT(sistema.no_sistema, '/', perfil.no_perfil, '/', modulo.no_modulo, '/', funcionalidade.no_funcionalidade) AS no_permissao
			FROM 
				seguranca.tb_permissao permissao
				INNER JOIN seguranca.tb_sistema sistema ON sistema.id_sistema = permissao.id_sistema
				INNER JOIN seguranca.tb_perfil perfil ON perfil.id_perfil = permissao.id_perfil
				INNER JOIN seguranca.tb_modulo modulo ON modulo.id_modulo = permissao.id_modulo
				INNER JOIN seguranca.tb_funcionalidade funcionalidade ON funcionalidade.id_funcionalidade = permissao.id_funcionalidade
			WHERE
				permissao.ic_ativo = 1
				$filter
			ORDER BY
				no_permissao
		";
		$rst = new TbUsuarioPermissao;
		return new \Phalcon\Mvc\Model\Resultset\Simple(null, $rst, $rst->getReadConnection()->query($phql, $bind));
```

This was the only way I found to make it work in this new version.

If you find a way to work as before let me know.

**Comment by GiTnEd** [comment]
Hello,
Experiencing same issue with Phalcon 2.0.2 upgrade from 1.3.4

**Select Tag creation issue on Resultset with Concat Columns (custom created field from query)**
Even when re-written using a form class extension - end with the same results
Posted here is the simpler tag->select used in the controller

### Symptom

Select Box is Empty when resultset contains a new Concatenated Column
this method worked previously in 1.3.4

### Input
```php
$columns = "CONCAT(inspect_group,' : ',name) as inspect, id";

$inspect_find = Inspections::find(array('columns' => $columns));
                
$inspection_select = $this->tag->select(
	array(  'id', 
                    $inspect_find, 
                    'using' => array('id', 'inspect'),
                    'class' => 'form-control',
                    'useEmpty' => true,
                    'emptyText' => 'Select to Edit...',
                    'emptyValue' => ''
	      ));

$this->view->setVars(array('inspection_select' => $inspection_select )); 
```
### Output
```html
<select id="id" class="form-control" name="id">
    <option value=""></option>
    <option value="" selected="selected"></option>
    <option value="" selected="selected"></option>
    <option value="" selected="selected"></option>
</select>
```
### Standard Find

This works as expected with simple find – using column names from table
```php
$inspect_find = Inspections::find();

$inspection_select = $this->tag->select(
	array(  'id', 
$inspect_find, 
'using' => array('id', 'name'),
'class' => 'form-control',
'useEmpty' => true,
'emptyText' => 'Select to Edit...',
'emptyValue' => ''
	));
```

### Output – Standard Find
```html
<select id="id" class="form-control" name="id">
    <option value="">Select to Edit...</option>
    <option value="1">Name1</option>
    <option value="2">Name2</option>
    <option value="3">Name3</option>
</select>
```

Seems to me - having to write a simple query and join things afterwards defeats the purpose if this was working in previous iterations of Phalcon.

**Comment by andresgutierrez** [comment]
This is fixed in the 2.0.x branch
